core/common/boot/src/main/java/org/eclipse/edc/boot/system/SyntheticExtension.java

```diff
@@ -0,0 +1,26 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.boot.system;
+
+import org.eclipse.edc.runtime.metamodel.annotation.Provides;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.telemetry.Telemetry;
+import org.eclipse.edc.spi.types.TypeManager;
+
+import java.time.Clock;
+
+@Provides({ Monitor.class, TypeManager.class, Telemetry.class, Clock.class })
```

Comment:
I get why `Monitor` and `Telemetry` should be "peculiar", as they are a specific loading mode in the `ExtensionLoader`, but `TypeManager` and `Clock` could be treated as standard services, or am I wrong?
This could be something for another issue maybe.

Reply:
All of them are registered in the `DefaultServiceEXtensionContext`'s constructor, i.e. they are not provided by an extension. That is why I had intended to treat them equally. Also, the original problem @FlorianSeidelAWS reported dealt with a potential cyclic dependency, which would hit us just the same with the `TM` and the `Clock`

Reply:
Ok, but there's no actual need to have `TypeManager` and `Clock` registered before the service loading and passed to the context. Will create an issue about that, nothing to do here.

Reply:
@ndr-brt the only think to be aware of is that in case they get provided conventionally by an extension, that particular extension should have as little incoming dependencies (`@Inject`) as possible to minimize the risk for dependency cycles

Reply:
that extension should be located in `boot`, so at the root of everything

Reply:
> that extension should be located in `boot`, so at the root of everything

I meant "code" dependencies, not Gradle dependencies. Ideally it should not have any `@Inject`-ed fields at all, which would make it dependent on other extensions, which we should avoid at all cost - only then is it truly at the "root" (=beginning) of the topologically sorted list.